bundle/config/mutator/apply_presets.go

```diff
@@ -221,6 +222,19 @@ func (m *applyPresets) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnos
 		dashboard.DisplayName = prefix + dashboard.DisplayName
 	}
 
+	if config.IsExplicitlyEnabled((b.Config.Presets.InPlaceDeployment)) {
+		root := b.SyncRoot.Native()
+		_, ok := env.Lookup(ctx, envDatabricksRuntimeVersion)
+		isInWorkspace := ok && strings.HasPrefix(root, "/Workspace/")
+
+		if isInWorkspace {
+			b.Config.Workspace.FilePath = b.BundleRootPath
```

Comment:
This should be `b.SyncRootPath` instead of `b.BundleRootPath`. We allow users to configure a parent directory as the upload destination.